### ISSUE TITLE
RuntimeErrors when running in docker

### ISSUE THREAD

**Issue by jerrylususu** [baxtree/subaligner#56]
### Describe the bug
When using Subaligner in dual mode for a short audio file (6min) inside docker, it timed out with multiple Python RuntimeError exception and failed to generate an aligned subtitle.

### To Reproduce
> The audio file `recording_CS102E_140.wav` and the subtitle file `new.srt` is placed in the given path (`c/Temporary/audio_text_alignment`), but are later copied to home directory for better IO perfor

1. Run `docker run -v "/c/Temporary/audio_text_alignment":/media -w "/media" -it baxtree/subaligner bash`
2. Run `cp new.srt  recording_CS102E_140.wav ~/`
3. Run `cd ~`
2. Run `subaligner -m dual -v recording_CS102E_140.wav -s new.srt`

### Expected behavior
Generate an aligned subtitle file.

### Screenshots and logs

![image](https://user-images.githubusercontent.com/17522475/114234454-73f4f480-9987-11eb-9979-05a418ff994d.png)

```
subaligner.predictor - ERROR - MainThread - Segment alignment timed out after 60 seconds
RuntimeError: module compiled against API version 0xe but this version of numpy is 0xd
subaligner.predictor - INFO - ThreadPoolExecutor-0_0 - [ThreadPoolExecutor-0_0] Segment 0 stretched
  File "/usr/local/lib/python3.8/dist-packages/subaligner/__main__.py", line 196, in main
    aligned_subs, subs, voice_probabilities, frame_rate = predictor.predict_dual_pass(
  File "/usr/local/lib/python3.8/dist-packages/subaligner/predictor.py", line 122, in predict_dual_pass
    new_subs = self.__predict_2nd_pass(
  File "/usr/local/lib/python3.8/dist-packages/subaligner/predictor.py", line 302, in __predict_2nd_pass
    raise TerminalException(message) from e
```

[output.log](https://github.com/baxtree/subaligner/files/6288141/output.log)
[stderr.log](https://github.com/baxtree/subaligner/files/6288142/stderr.log)
[stdout.log](https://github.com/baxtree/subaligner/files/6288144/stdout.log)

### Media files
[audio_text_alignment.zip](https://github.com/baxtree/subaligner/files/6288179/audio_text_alignment.zip)
Note: see additional context

### Environment
 - Operating System: Windows 10 x64 20H2 (with Docker Desktop v3.2.2(61853), Engine 20.10.5)
 - Python Version: 3.8.6 (checked from inside docker)
 - CUDA: No cuda support on machine
 - Docker Image info
     ```
    REPOSITORY           TAG       IMAGE ID       CREATED        SIZE
    baxtree/subaligner   latest    580481074d20   3 days ago     2.92GB
    ```

### Additional context
The task I am having is that I need to split a large sentence into multiple smaller ones, as I describe in #49 . To do that, I simply cut the sentence based on length and remained the start and end timestamp. The one named `recording_CS102E_140.srt` is the original one and the one named `new.srt` is the one I manually cut.

**Comment by jerrylususu** [comment]
Just tried `pip3 install numpy --upgrade`, but it seems numpy 1.20 is not even competible, giving out some `NotImplementedError`s.

![image](https://user-images.githubusercontent.com/17522475/114239581-ec12e880-998e-11eb-92fc-6adb4c0c79d9.png)

```
Cannot convert a symbolic Tensor (bidirectional/forward_lstm/strided_slice:0) to a numpy array. This error may indicate that you're trying to pass a Tensor to a NumPy call, which is not supported

  File "/usr/local/lib/python3.8/dist-packages/subaligner/__main__.py", line 196, in main
    aligned_subs, subs, voice_probabilities, frame_rate = predictor.predict_dual_pass(
  File "/usr/local/lib/python3.8/dist-packages/subaligner/predictor.py", line 114, in predict_dual_pass
    self.__initialise_network(weights_dir)
  File "/usr/local/lib/python3.8/dist-packages/subaligner/predictor.py", line 455, in __initialise_network
    self.__network = Network.get_from_model(model_path, hyperparams)
  File "/usr/local/lib/python3.8/dist-packages/subaligner/network.py", line 132, in get_from_model
    return cls(
  File "/usr/local/lib/python3.8/dist-packages/subaligner/network.py", line 80, in __init__
    self.__model = load_model(model_path)
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/keras/saving/save.py", line 206, in load_model
    return hdf5_format.load_model_from_hdf5(filepath, custom_objects,
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/keras/saving/hdf5_format.py", line 183, in load_model_from_hdf5
    model = model_config_lib.model_from_config(model_config,
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/keras/saving/model_config.py", line 64, in model_from_config
    return deserialize(config, custom_objects=custom_objects)
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/keras/layers/serialization.py", line 173, in deserialize
    return generic_utils.deserialize_keras_object(
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/keras/utils/generic_utils.py", line 354, in deserialize_keras_object
    return cls.from_config(
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/keras/engine/training.py", line 2261, in from_config
    return functional.Functional.from_config(
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/keras/engine/functional.py", line 668, in from_config
    input_tensors, output_tensors, created_layers = reconstruct_from_config(
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/keras/engine/functional.py", line 1285, in reconstruct_from_config
    process_node(layer, node_data)
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/keras/engine/functional.py", line 1233, in process_node
    output_tensors = layer(input_tensors, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/keras/layers/wrappers.py", line 539, in __call__
    return super(Bidirectional, self).__call__(inputs, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/keras/engine/base_layer.py", line 951, in __call__
    return self._functional_construction_call(inputs, args, kwargs,
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/keras/engine/base_layer.py", line 1090, in _functional_construction_call
    outputs = self._keras_tensor_symbolic_call(
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/keras/engine/base_layer.py", line 822, in _keras_tensor_symbolic_call
    return self._infer_output_signature(inputs, args, kwargs, input_masks)
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/keras/engine/base_layer.py", line 863, in _infer_output_signature
    outputs = call_fn(inputs, *args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/keras/layers/wrappers.py", line 652, in call
    y = self.forward_layer(forward_inputs,
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/keras/layers/recurrent.py", line 660, in __call__
    return super(RNN, self).__call__(inputs, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/keras/engine/base_layer.py", line 1012, in __call__
    outputs = call_fn(inputs, *args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/keras/layers/recurrent_v2.py", line 1157, in call
    inputs, initial_state, _ = self._process_inputs(inputs, initial_state, None)
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/keras/layers/recurrent.py", line 859, in _process_inputs
    initial_state = self.get_initial_state(inputs)
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/keras/layers/recurrent.py", line 642, in get_initial_state
    init_state = get_initial_state_fn(
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/keras/layers/recurrent.py", line 2506, in get_initial_state
    return list(_generate_zero_filled_state_for_cell(
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/keras/layers/recurrent.py", line 2987, in _generate_zero_filled_state_for_cell
    return _generate_zero_filled_state(batch_size, cell.state_size, dtype)
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/keras/layers/recurrent.py", line 3003, in _generate_zero_filled_state
    return nest.map_structure(create_zeros, state_size)
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/util/nest.py", line 659, in map_structure
    structure[0], [func(*x) for x in entries],
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/util/nest.py", line 659, in <listcomp>
    structure[0], [func(*x) for x in entries],
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/keras/layers/recurrent.py", line 3000, in create_zeros
    return array_ops.zeros(init_state_size, dtype=dtype)
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/util/dispatch.py", line 201, in wrapper
    return target(*args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/ops/array_ops.py", line 2819, in wrapped
    tensor = fun(*args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/ops/array_ops.py", line 2868, in zeros
    output = _constant_if_small(zero, shape, dtype, name)
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/ops/array_ops.py", line 2804, in _constant_if_small
    if np.prod(shape) < 1000:
  File "<__array_function__ internals>", line 5, in prod
  File "/usr/local/lib/python3.8/dist-packages/numpy/core/fromnumeric.py", line 3030, in prod
    return _wrapreduction(a, np.multiply, 'prod', axis, dtype, out,
  File "/usr/local/lib/python3.8/dist-packages/numpy/core/fromnumeric.py", line 87, in _wrapreduction
    return ufunc.reduce(obj, axis, dtype, out, **passkwargs)
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/framework/ops.py", line 852, in __array__
    raise NotImplementedError(
```

Also tried `pip3 install numpy==1.19.2`, yet still get `RuntimeError: module compiled against API version 0xe but this version of numpy is 0xd`. Seems no luck.
Also for the note here, the version of numpy bundled with the docker image seems to be 1.19.5 .

**Comment by jerrylususu** [comment]
Tried an older image (0.1.12.u20), but the error is still there.

**Comment by jerrylususu** [comment]
Tried to install subaligner on a fresh new ubuntu:20.04 image, yet still fails. Same error as the latest image.

**Comment by baxtree** [comment]
Hi, @jerrylususu, Thanks for reporting this. Apparently, there has been a regression on subaligner's ubuntu:20.04 image (latest). For the time being, the workaround is to use one of the other images such as ubuntu:18.04:
`docker pull baxtree/subaligner:0.1.3.u18`

**Comment by jerrylususu** [comment]
Thanks for your timely response. I just tried `0.1.3.u18` and it seems to fix the issue. The results are satisfactory and thanks for your outstanding work!

**Comment by baxtree** [comment]
Hi, @jerrylususu , both `latest` and `0.1.3.u20` images should have been fixed by #57. Please have it a go if you want to get rid of the above workaround. Thanks again for reporting this issue.

**Comment by jerrylususu** [comment]
Thanks for updating! I just checked out the current version (using latest tag) and it is working correctly as expected.
